### PR TITLE
Docker - RNA-seq

### PR DESCRIPTION
**Pull request by bgruening**
This PR demonstrates how we can easily pre-populate Docker images with reference indices.

ping @nekrut, @bebatut, @MoHeydarian, @malloryfreeberg

### CODE REVIEW

File: RNA-Seq/docker/Dockerfile [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
 
 MAINTAINER Björn A. Grüning, bjoern.gruening@gmail.com
 
 ENV GALAXY_CONFIG_BRAND RNA-Seq training
-ENV ENABLE_TTS_INSTALL True
 
-# Enable Conda dependency resolution
-ENV GALAXY_CONFIG_CONDA_AUTO_INSTALL=True \
-    GALAXY_CONFIG_CONDA_AUTO_INIT=True
+ADD fetch_and_index_genomes.yaml data_manager_rna_seq.yaml
```
Comment (nsoranzo) [inline RNA-Seq/docker/Dockerfile]: Where does `fetch_and_index_genomes.yaml` come from?

**Comment by bgruening** [comment]
Related PR: https://github.com/galaxyproject/galaxy/pull/3874

File: RNA-Seq/docker/Dockerfile (hunk 1 above)
Comment (bgruening) [inline RNA-Seq/docker/Dockerfile]: nowhere, I renamed this in the last minute to be more meaningful.

**Comment by bgruening** [comment]
If everyone agrees I would like to put the images under the `biocontainers` namespace. This one will be build here: biocontainers/galaxy-rna-seq-training

**Comment by malloryfreeberg** [comment]
Just to make sure I'm understanding correctly, a `data_manager_rna_seq.yaml` is made that lists the genome indices that the user wants to have loaded into the Docker image?

**Comment by bgruening** [comment]
@malloryfreeberg yes exactly. You can define the genome of interest and the indices that you want to have installed into the Docker container with this file - as long as there is a data-manager installed.

**Comment by malloryfreeberg** [comment]
@bgruening This is a life-saver. Thanks so much!

**Comment by bgruening** [comment]
@malloryfreeberg it can be that still tools are missing in this container, if this is the case please ping me and I will add them or show you howto add them. tools, data, tours, vis should be all possible now very easily.

**Comment by malloryfreeberg** [comment]
@bgruening If this is supposed to cover all the RNA-seq tutorials, then `data_manager_rna_seq.yaml` also missing mm10 indices, which are needed for the de novo RNA-seq tutorial. I can add that. Where can we add tools that are missing?

File: RNA-Seq/docker/data_manager_rna_seq.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      params:
+        - 'dbkey_source|dbkey': '{{ item }}'
+        - 'reference_source|reference_source_selector': 'ucsc'
+        - 'reference_source|requested_dbkey': '{{ item }}'
+      # Items refere to a list of variables you want to run this data manager. You can use them inside the param field with {{ item }}
+      # In case of genome for example you can run this DM with multiple genomes, or you could give multiple URLs.
+      items:
+        #- ce8
+        - dm3
+        #- mm9
```
Comment (malloryfreeberg) [inline RNA-Seq/docker/data_manager_rna_seq.yaml]: Need mm10 for de novo RNA-seq tutorial

File: RNA-Seq/docker/data_manager_rna_seq.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        - all_fasta
+        - __dbkeys__
+
+    - id: toolshed.g2.bx.psu.edu/repos/devteam/data_manager_bowtie2_index_builder/bowtie2_index_builder_data_manager/2.2.6
+      params:
+        - 'all_fasta_source': '{{ item }}'
+      items:
+        #- ce8
+        - dm3
+        #- mm9
```
Comment (malloryfreeberg) [inline RNA-Seq/docker/data_manager_rna_seq.yaml]: Need mm10 for de novo RNA-seq tutorial

File: RNA-Seq/docker/data_manager_rna_seq.yaml (hunk 2 above)
Comment (bgruening) [inline RNA-Seq/docker/data_manager_rna_seq.yaml]: @malloryfreeberg do we need dm3 at all? This was just a demonstration ... :)

Comment (malloryfreeberg) [inline RNA-Seq/docker/data_manager_rna_seq.yaml]: Yes, keep dm3 because it is used for the ref-based RNA-seq tutorial :)

**Comment by bgruening** [comment]
@malloryfreeberg adding tools might need a little bit of coordination :)

This training image is based on the RNA-seq Docker image (I thought this is a nice idea). So if you think this tool belongs to a general RNA-seq image add it here: https://github.com/bgruening/galaxy-rna-seq/blob/master/rna_seq_tools.yml

The RNA-seq Image in turn is based on the NGS-preprocessing image, which is defined here: https://github.com/bgruening/docker-galaxy-ngs-preprocessing
This image is also the base for the ChIP-Seq image and the Exome-Seq image and so on. So if your tool is so general that it belongs somehow to prepossessing add it there - and it will pop-up in all the other flavors that build on top of https://github.com/bgruening/docker-galaxy-ngs-preprocessing.

If you think your tool is only very useful for the training create a new YAML file like I linked above and push it next to the Dockerfile :)

Not sure if this is to complicated - but with this we can share as many resources as possible.

**Comment by bgruening** [comment]
@malloryfreeberg there is now an image: `docker pull biocontainers/galaxy-rna-seq-training`
It should container mm10 and dm3, let me know what you think.

**Pull request merged**